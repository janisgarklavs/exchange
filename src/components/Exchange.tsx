import React from "react";

import { ExchangeRates } from "./ExchangeRates";

import { Exchange as IExchange } from "../store/exchange/types";

interface ExchangeProps {
  data: IExchange;
}

export const Exchange: React.SFC<ExchangeProps> = ({ data }) => (
  <>
    <div
      style={{
        display: "flex",
        justifyContent: "space-between",
        borderBottom: "1px solid black",
        padding: "12px 24px"
      }}
    >
      <div style={{ fontWeight: "bold" }}>
        Base: <span>{data.base}</span>
      </div>
      <div style={{ fontWeight: "bold" }}>
        Date: <span>{data.date}</span>
      </div>
    </div>

    <ExchangeRates rates={data.rates} />
  </>
);
