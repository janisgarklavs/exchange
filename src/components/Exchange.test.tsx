import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Exchange } from "./Exchange";

describe("<Exchange />", () => {
  test("it renders correctly", () => {
    const data = {
      date: "01-01-2018",
      base: "EUR",
      rates: { USD: 0.2 }
    };
    const component = shallow(<Exchange data={data} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
