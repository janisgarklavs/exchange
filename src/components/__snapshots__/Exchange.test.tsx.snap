// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exchange /> it renders correctly 1`] = `
<Fragment>
  <div
    style={
      Object {
        "borderBottom": "1px solid black",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "12px 24px",
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Base: 
      <span>
        EUR
      </span>
    </div>
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Date: 
      <span>
        01-01-2018
      </span>
    </div>
  </div>
  <ExchangeRates
    rates={
      Object {
        "USD": 0.2,
      }
    }
  />
</Fragment>
`;
